python/hyperon/atoms.py

```diff
@@ -250,3 +253,62 @@ def iterator(self):
             result.append((r[0], Atom._from_catom(r[1])))
 
         return iter(result)
+
+class BindingsSet:
+
+    def __init__(self, input: Union[hp.CBindingsSet, Bindings, None] = None):
+        if input is None:
+            self.c_set = hp.bindings_set_single()
+        elif isinstance(input, Bindings):
+            self.c_set = hp.bindings_set_from_bindings(input.cbindings)
+        else:
+            self.c_set = input
+
+    def __del__(self):
+        if self.c_set is not None:
+            hp.bindings_set_free(self.c_set)
+            self.c_set = None
+
+    def __eq__(self, other):
+        return (isinstance(other, BindingsSet) and
+                hp.bindings_set_eq(self.c_set, other.c_set))
+
+    def __repr__(self):
+        return hp.bindings_set_to_str(self.c_set)
+
+    def __deepcopy__(self, memodict={}):
+        return self.clone()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.c_set is not None:
+            hp.bindings_set_free(self.c_set)
+            self.c_set = None
+
+    def clone(self):
+        return BindingsSet(hp.bindings_set_clone(self.c_set))
+
+    def is_empty(self) -> bool:
+        return hp.bindings_set_is_empty(self.c_set)
+
+    def add_var_binding(self, var_name:str, atom: Atom) -> bool:
```

Comment:
I would made it an Atom, as far as you introduced it on the C API level. Same for the `add_var_equality`.

Reply:
Done.  I also added a union type argument to Bindings.add_var_bindings() in the Python code, so it's possible to interact with Bindings and BindingsSets in a consistent way from Python.